src/extension.ts

```diff
@@ -322,8 +322,19 @@ async function sourceRosAndWorkspace(): Promise<void> {
     let attemptWorkspaceDiscovery = true;
 
     if (rosSetupScript) {
+        // Regular expression to match '${workspaceFolder}'
+        const regex = "\$\{workspaceFolder\}";
+        if (rosSetupScript.includes(regex)) {
+            if (vscode.workspace.workspaceFolders.length === 1) {
+                // Replace all occurrences of '${workspaceFolder}' with the workspace string
+                rosSetupScript = rosSetupScript.replace(regex, vscode.workspace.workspaceFolders[0].uri.fsPath);
+            } else {
+                vscode.window.showErrorMessage(`Multiple or No workspaces found, but the ROS setup script setting \"ros.rosSetupScript\" is configured with '${rosSetupScript}'`);
```

Comment:
Thank you for your contribution! 
Since this workflow can happen outside of a ROS Workspace, having this as a dialog may be a bit jarring.
Could you change it to use `outputChannel.appendLine`;

Reply:
naturally! should be done.